### PR TITLE
topo: cache CellInfo along with conn

### PR DESCRIPTION
**Pull request by rsajwani**
Co-authored-by: Rameez Sajwani <rameezwazirali@hotmail.com>
Signed-off-by: Rameez Sajwani <rameezwazirali@hotmail.com>

## Description

Issue: vtctlclient ListAllTablets does not show those tablets in cells that were deleted and re-added with same name.
The root cause is that we cache connections to cells by name, and the name is being re-used.

We are fixing this by caching the CellInfo along with conn so that we can verify that it is the exact "same" cell before using the cached connection.

## Repro

We create a cell "abc" with some rootPath like path1/abc. Then we delete it, and create it again with the same name but a different path like path2/abc.
The vtctld cache will continue to contains a connection with path path1/abc which no longer exists.

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: test/config.json [hunk 1]
```diff
@@ -1228,6 +1228,24 @@
...
+		},
+		"vreplication_source_time_zone": {
+			"File": "unused.go",
+			"Args": ["vitess.io/vitess/go/test/endtoend/vreplication", "-run", "TestMoveTablesTZ"],
+			"Command": [],
+			"Manual": false,
+			"Shard": "vreplication_migrate",
+			"RetryMax": 1,
+			"Tags": []
+		},
```
Comment (deepthi) [inline test/config.json:1240]: This change seems unrelated, maybe it should not be in this PR?

Comment (rsajwani) [reply test/config.json:1240]: yeah looks like it added as part of auto conflict resolution .. let me remove it.

File: .github/workflows/cluster_endtoend_topo_connection_cache.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    - name: Configure git private repo access
+      env:
+        GITHUB_TOKEN: ${{ secrets.PLANETSCALE_ACTIONS_BOT_TOKEN }}
+      run: |
+        git config --global --add url."https://${GITHUB_TOKEN}@github.com/".insteadOf "https://github.com/"
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.17.9
```
Comment (frouioui) [inline .github/workflows/cluster_endtoend_topo_connection_cache.yml:24]: This should be at least `go.1.18.1`

File: .github/workflows/cluster_endtoend_topo_connection_cache.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# DO NOT MODIFY: THIS FILE IS GENERATED USING "make generate_ci_workflows"
+
+name: Cluster (topo_connection_cache)
+on: [push]
```
Comment (frouioui) [inline .github/workflows/cluster_endtoend_topo_connection_cache.yml:4]: You should add `pull_request` to the list of targets as well. Otherwise, the build will never start on a "regular" commit that lives in a Pull Request. The target `push` only works when a commit is pushed on any branch of `vitessio/vitess` (no its forks).

File: go/vt/topo/server.go [hunk 4]
```diff
@@ -56,7 +58,7 @@ import (
 
 const (
 	// GlobalCell is the name of the global cell.  It is special
-	// as it contains the global topology, and references the other cells.
+	// as it contains the global topology, and references the other cellConns.
```
Comment (deepthi) [inline go/vt/topo/server.go:61]: This change should be reverted.

File: go/vt/topo/server.go [hunk 5]
```diff
@@ -98,7 +100,7 @@ type Factory interface {
 	// HasGlobalReadOnlyCell returns true if the global cell
 	// has read-only replicas of the topology data. The global topology
 	// is usually more expensive to read from / write to, as it is
-	// replicated over many cells. Some topology services provide
+	// replicated over many cellConns. Some topology services provide
```
Comment (deepthi) [inline go/vt/topo/server.go:103]: This should also be reverted.

Comment (deepthi) [inline go/vt/topo/server.go:103]: This should also be reverted.

File: go/vt/topo/server.go [hunk 6]
```diff
@@ -234,46 +241,46 @@ func Open() *Server {
 }
 
 // ConnForCell returns a Conn object for the given cell.
-// It caches Conn objects from previously requested cells.
+// It caches Conn objects from previously requested cellConns.
```
Comment (deepthi) [inline go/vt/topo/server.go:244]: This should also be reverted.

Comment (deepthi) [inline go/vt/topo/server.go:244]: This should also be reverted.

**Review by deepthi — APPROVED** [review]
